File: 0062-unique-paths/0062-unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        def factorial(k:int)->int:
            fac:int = 1
            for i in range(1,k+1):
                fac *=i
            return fac
        return (factorial(m+n-2)//factorial(m-1))//factorial(n-1)       

File: 0062-unique-paths/0062-unique-paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int>dp(n+1, 1);
        for(int i=2; i<=m; i++){
            for(int j=2; j<=n; j++){
                dp[j] = dp[j]+dp[j-1];
            }
        }
        return dp[n];
    }
};